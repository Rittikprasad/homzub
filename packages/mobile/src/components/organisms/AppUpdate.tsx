/**
 * Handling Force Update, Maintenance logic using remote config
 * Doc for ref: https://rnfirebase.io/remote-config/usage
 */

import React, { useEffect, useState } from 'react';
import { Linking, StyleSheet, View } from 'react-native';
import remoteConfig from '@react-native-firebase/remote-config';
import { AppModes, ConfigHelper, ConfigModes } from '@homzhub/common/src/utils/ConfigHelper';
import { DeviceUtils } from '@homzhub/common/src/utils/DeviceUtils';
import { Logger } from '@homzhub/common/src/utils/Logger';
import { PlatformUtils } from '@homzhub/common/src/utils/PlatformUtils';
import { I18nService } from '@homzhub/common/src/services/Localization/i18nextService';
import Rocket from '@homzhub/common/src/assets/images/rocket.svg';
import { theme } from '@homzhub/common/src/styles/theme';
import { Button } from '@homzhub/common/src/components/atoms/Button';
import { Label } from '@homzhub/common/src/components/atoms/Text';
import { BottomSheet } from '@homzhub/common/src/components/molecules/BottomSheet';

// Remote config Parameters
enum ConfigKeys {
  MINIMUM_VERSION = 'minimum_version',
  LATEST_VERSION = 'latest_version',
  IS_UNDER_MAINTENANCE = 'is_under_maintenance',
  MESSAGES = 'messages',
}

interface IMessages {
  update: string;
  maintenance: string;
}

interface IConfig {
  [key: string]: any;
}

interface IOwnState {
  isForceUpdate: boolean;
  isUpdateAvailable: boolean;
  isUnderMaintenance: boolean;
  messages: IMessages;
}

const initialState: IOwnState = {
  isForceUpdate: false,
  isUpdateAvailable: false,
  isUnderMaintenance: false,
  messages: {
    update: '',
    maintenance: '',
  },
};

const AppUpdate = (): React.ReactElement => {
  const [configData, setConfigData] = useState<IConfig[]>([]);
  const [configValues, setConfigValues] = useState<IOwnState>(initialState);
  const [isSheetVisible, setVisibility] = useState(false);

  useEffect(() => {
    remoteConfig()
      .setConfigSettings({
        minimumFetchIntervalMillis: 30000, // Global Caching frequency
      })
      .then(() => {
        // Set Default before fetching config from firebase
        remoteConfig()
          .setDefaults({
            is_under_maintenance: false,
            latest_version: '',
            minimum_version: '',
          })
          .then(() => remoteConfig().fetchAndActivate()) // Fetch config and Activate
          .then(() => {
            const config = remoteConfig().getAll(); // Get All config list

            if (config) {
              const formattedData = Object.entries(config).map((item) => {
                const [key, entry] = item as any;
                return {
                  [key]: JSON.parse(entry.asString()),
                };
              });
              setConfigData(formattedData);
            }
          })
          .then(() => remoteConfig().fetch()); // Fetching data for next process
      })

      .catch((e) => {
        Logger.warn(e.message);
      });
  }, []);

  useEffect(() => {
    const updatedData = { ...configValues };
    const configMode = ConfigHelper.getConfigMode();
    const appMode = (configMode === ConfigModes.PROD ? AppModes.RELEASE : AppModes.DEBUG).toLowerCase();

    // Structuring config and maintaining local state to handle UI
    configData.forEach((item) => {
      if (Object.keys(item)[0] === ConfigKeys.MINIMUM_VERSION) {
        updatedData.isForceUpdate = item[ConfigKeys.MINIMUM_VERSION][appMode] > DeviceUtils.appVersion;
      } else if (Object.keys(item)[0] === ConfigKeys.LATEST_VERSION) {
        updatedData.isUpdateAvailable = item[ConfigKeys.LATEST_VERSION][appMode] > DeviceUtils.appVersion;
      } else if (Object.keys(item)[0] === ConfigKeys.IS_UNDER_MAINTENANCE) {
        updatedData.isUnderMaintenance = item[ConfigKeys.IS_UNDER_MAINTENANCE][appMode];
      } else {
        updatedData.messages = item[ConfigKeys.MESSAGES];
      }
    });

    const isVisible = updatedData.isUpdateAvailable || updatedData.isUnderMaintenance || updatedData.isForceUpdate;

    setVisibility(isVisible);
    setConfigValues(updatedData);
  }, [configData]);

  const onUpdateNow = async (): Promise<void> => {
    const { getAppleStoreUrl, getGooglePlayStoreUrl } = ConfigHelper;
    await Linking.openURL(PlatformUtils.isIOS() ? getAppleStoreUrl() : getGooglePlayStoreUrl());
  };

  const { isForceUpdate, messages, isUnderMaintenance, isUpdateAvailable } = configValues;
  const isAlwaysVisible = isForceUpdate || isUnderMaintenance;
  return (
    <BottomSheet
      visible={isSheetVisible}
      isShadowView
      headerTitle={isUpdateAvailable ? I18nService.t('common:timeToUpdate') : I18nService.t('common:underMaintenance')}
      sheetHeight={isUpdateAvailable ? 420 : 300}
      isCloseOnDrag={!isAlwaysVisible}
      isAlwaysVisible={isAlwaysVisible}
      onCloseSheet={(): void => setVisibility(false)}
    >
      <View style={styles.container}>
        <Rocket style={styles.iconStyle} />
        <Label type="large" style={styles.message}>
          {isUpdateAvailable ? messages?.update : messages?.maintenance}
        </Label>
        {!isUnderMaintenance && (
          <Button
            type="primary"
            title={I18nService.t('common:updateNow')}
            containerStyle={styles.updateButton}
            onPress={onUpdateNow}
          />
        )}
        {!isAlwaysVisible && (
          <Button
            type="secondaryOutline"
            title={I18nService.t('common:doLater')}
            containerStyle={styles.secondaryButton}
            textStyle={styles.buttonTitle}
            onPress={(): void => setVisibility(false)}
          />
        )}
      </View>
    </BottomSheet>
  );
};

export default AppUpdate;

const styles = StyleSheet.create({
  container: {
    padding: 32,
    flex: 1,
  },
  iconStyle: {
    alignSelf: 'center',
  },
  message: {
    textAlign: 'center',
    marginVertical: 16,
    color: theme.colors.darkTint3,
  },
  updateButton: {
    flex: 0,
    height: 46,
    marginVertical: 10,
  },
  secondaryButton: {
    height: 46,
    flex: 0,
  },
  buttonTitle: {
    color: theme.colors.darkTint3,
  },
});
