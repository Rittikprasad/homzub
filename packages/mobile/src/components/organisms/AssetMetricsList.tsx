import React, { useState, useCallback } from 'react';
import { StyleProp, StyleSheet, TouchableOpacity, View, ViewStyle } from 'react-native';
import { useTranslation } from 'react-i18next';
import { useRoute } from '@react-navigation/native';
import { useNavigation } from '@react-navigation/core';
import { theme } from '@homzhub/common/src/styles/theme';
import Icon, { icons } from '@homzhub/common/src/assets/icon';
import Logo from '@homzhub/common/src/assets/images/homzhubDashboard.svg';
import { Label, Text } from '@homzhub/common/src/components/atoms/Text';
import { PaginationComponent } from '@homzhub/mobile/src/components/atoms/PaginationComponent';
import { SnapCarousel } from '@homzhub/mobile/src/components/atoms/Carousel';
import { AssetMetrics } from '@homzhub/mobile/src/components/molecules/AssetMetrics';

export interface IMetricsData {
  name: string;
  count: string | number;
  label?: string;
  id?: number;
  isCurrency?: boolean;
  colorCode: string;
  code?: string;
}

interface IProps {
  data: IMetricsData[];
  title?: string;
  subscription?: string;
  selectedAssetType?: string;
  numOfElements?: number;
  isSubTextRequired?: boolean;
  onPlusIconClicked?: () => void;
  onMetricsClicked?: (name: string) => void;
  containerStyle?: StyleProp<ViewStyle>;
  subTitleText?: string;
  headerIcon?: string;
  showBackIcon?: boolean;
}

const COMPONENT_PADDING = 12;
const SLIDER_WIDTH = theme.viewport.width - theme.layout.screenPadding * 2;
const AssetMetricsList = (props: IProps): React.ReactElement => {
  const {
    title = 0,
    data,
    selectedAssetType,
    onPlusIconClicked,
    containerStyle,
    onMetricsClicked,
    numOfElements = 3,
    isSubTextRequired = true,
    subTitleText,
    headerIcon,
    showBackIcon = false,
  } = props;

  // HOOKS
  const { t } = useTranslation();
  const { goBack, setParams } = useNavigation();
  const { params } = useRoute();
  const [activeIndex, setActiveIndex] = useState(0);
  // HELPERS
  const bubblePlusIcon = useCallback((): void => {
    if (onPlusIconClicked) {
      onPlusIconClicked();
    }
  }, [onPlusIconClicked]);

  const splitData = useCallback((): IMetricsData[][] => {
    const newArr = [];

    for (let i = 0; i < data.length; i += numOfElements) {
      newArr.push(data.slice(i, i + numOfElements));
    }

    return newArr;
  }, [data, numOfElements]);

  const onGoBack = (): void => {
    goBack();
    // @ts-ignore
    if (params && params?.isFromNavigation) {
      setParams({ isFromNavigation: false });
    }
  };
  // HELPERS END

  const renderItem = useCallback(
    (items: IMetricsData[]): React.ReactElement => {
      return (
        <View style={styles.sliderRow}>
          {items.map((item: IMetricsData) => {
            const cardStyle = {
              minWidth: (SLIDER_WIDTH - COMPONENT_PADDING * (numOfElements + 1)) / numOfElements,
            };
            // TODO: (Shikha) - Connect with BE to handle all logic with CODE
            const handlePress = (): void => onMetricsClicked && onMetricsClicked(item.code || item.name);
            return (
              <AssetMetrics
                key={`${item.label ?? item.name}`}
                header={item.label ?? item.name}
                value={item.count}
                isCurrency={item.isCurrency ?? false}
                cardStyle={cardStyle}
                colorCode={item.colorCode}
                onPressMetrics={handlePress}
                selectedAssetType={selectedAssetType}
                // Todo : Check in multiple devices
                maxLength={items.length === 2 ? 15 : undefined}
              />
            );
          })}
        </View>
      );
    },
    [onMetricsClicked, numOfElements, selectedAssetType]
  );

  return (
    <View style={[styles.container, containerStyle]}>
      <View style={[styles.header, !isSubTextRequired && styles.financialView]}>
        {showBackIcon && (
          <TouchableOpacity onPress={onGoBack} style={styles.backIcon}>
            <Icon name={icons.leftArrow} size={25} color={theme.colors.blue} />
          </TouchableOpacity>
        )}
        {headerIcon ? (
          <View>
            <Icon name={headerIcon} size={30} style={styles.icon} color={theme.colors.blue} />
            <View style={[styles.headerIcon, styles.logo]} />
          </View>
        ) : (
          <Logo style={styles.logo} width={50} height={50} />
        )}
        <View style={styles.headerCenter}>
          <Text type="regular" textType="bold" style={styles.assetCount}>
            {title}
          </Text>
          <View style={styles.propertiesRow}>
            {isSubTextRequired && (
              <Label type="large" textType="semiBold" style={styles.propertyText}>
                {subTitleText || t('common:properties')}
              </Label>
            )}
          </View>
        </View>
        {onPlusIconClicked && (
          <TouchableOpacity onPress={bubblePlusIcon} style={styles.plusIconContainer}>
            <Icon name={icons.circularPlus} size={32} color={theme.colors.primaryColor} testID="icnPlus" />
          </TouchableOpacity>
        )}
      </View>
      <SnapCarousel
        carouselData={splitData()}
        carouselItem={renderItem}
        activeIndex={activeIndex}
        onSnapToItem={setActiveIndex}
        sliderWidth={SLIDER_WIDTH}
        itemWidth={SLIDER_WIDTH}
      />
      <PaginationComponent
        dotsLength={splitData().length}
        activeSlide={activeIndex}
        containerStyle={styles.paginationContainer}
        activeDotStyle={[styles.dot, styles.activeDot]}
        inactiveDotStyle={[styles.dot, styles.inactiveDot]}
      />
    </View>
  );
};

const memoizedComponent = React.memo(AssetMetricsList);
export { memoizedComponent as AssetMetricsList };

const styles = StyleSheet.create({
  container: {
    paddingVertical: 12,
    borderRadius: 4,
    backgroundColor: theme.colors.white,
  },
  header: {
    marginHorizontal: 12,
    flexDirection: 'row',
  },
  headerCenter: {
    flex: 1,
  },
  propertiesRow: {
    flexDirection: 'row',
  },
  sliderRow: {
    flexDirection: 'row',
  },
  paginationContainer: {
    paddingVertical: 0,
    marginTop: 12,
  },
  assetCount: {
    color: theme.colors.darkTint1,
  },
  propertyText: {
    color: theme.colors.darkTint4,
  },
  logo: {
    marginEnd: 12,
  },
  dot: {
    width: 8.5,
    height: 8.5,
  },
  activeDot: {
    borderWidth: 1.5,
  },
  inactiveDot: {
    backgroundColor: theme.colors.disabled,
    borderWidth: 0,
  },
  financialView: {
    alignItems: 'center',
  },
  headerIcon: {
    backgroundColor: theme.colors.blue,
    width: 50,
    height: 50,
    opacity: 0.1,
    borderRadius: 30,
  },
  icon: {
    position: 'absolute',
    left: 10,
    top: 10,
  },
  plusIconContainer: {
    justifyContent: 'center',
    padding: 5,
  },
  backIcon: {
    alignSelf: 'center',
    marginRight: 10,
  },
});
