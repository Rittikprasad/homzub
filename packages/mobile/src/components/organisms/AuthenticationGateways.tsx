import React from 'react';
import { StyleSheet, TouchableOpacity } from 'react-native';
import { connect } from 'react-redux';
import { bindActionCreators, Dispatch } from 'redux';
import { WithTranslation, withTranslation } from 'react-i18next';
import { StackNavigationProp } from '@react-navigation/stack';
import { ObjectMapper } from '@homzhub/common/src/utils/ObjectMapper';
import { PlatformUtils } from '@homzhub/common/src/utils/PlatformUtils';
import { SocialAuthUtils } from '@homzhub/common/src/utils/SocialAuthUtils';
import { AuthService } from '@homzhub/mobile/src/services/AuthService';
import { ScreensKeys } from '@homzhub/mobile/src/navigation/interfaces';
import Apple from '@homzhub/common/src/assets/images/apple.svg';
import Google from '@homzhub/common/src/assets/images/google.svg';
import Facebook from '@homzhub/common/src/assets/images/facebook.svg';
import LinkedIn from '@homzhub/common/src/assets/images/linkedin.svg';
import { UserActions } from '@homzhub/common/src/modules/user/actions';
import { Label } from '@homzhub/common/src/components/atoms/Text';
import CommonAuthenticationHoc from '@homzhub/common/src/components/organisms/CommonAuthenticationHOC';
import { AuthStackParamList } from '@homzhub/mobile/src/navigation/AuthStack';
import { SocialAuthProvider } from '@homzhub/common/src/domain/models/SocialAuthProvider';
import { ISocialUserData, SocialAuthKeys } from '@homzhub/common/src/constants/SocialAuthProviders';
import { IUserTokens } from '@homzhub/common/src/services/storage/StorageService';
import { LocaleConstants } from '@homzhub/common/src/services/Localization/constants';

interface IDispatchProps {
  loginSuccess: (data: IUserTokens) => void;
}

interface IAuthenticationGatewayProps {
  isFromLogin: boolean;
  onEmailLogin?: () => void;
  testID?: string;
  navigation?: StackNavigationProp<AuthStackParamList, ScreensKeys.SignUp | ScreensKeys.Login>;
  toggleLoading?: (isLoading: boolean) => void;
  onSuccessCallback?: () => void;
  isSocialAllowed?: boolean;
}

type Props = IAuthenticationGatewayProps & WithTranslation & IDispatchProps;

interface IOwnState {
  authProviders: SocialAuthProvider[];
}

class AuthenticationGateways extends React.PureComponent<Props, IOwnState> {
  public state = {
    authProviders: [],
  };

  public componentDidMount(): void {
    SocialAuthUtils.fetchSocialMedia((response) => {
      if (PlatformUtils.isIOS()) {
        response.unshift(
          ObjectMapper.deserialize(SocialAuthProvider, {
            provider: SocialAuthKeys.Apple,
            description: 'Apple',
            visible: PlatformUtils.isIOS(),
          })
        );
      }

      this.setState({ authProviders: response });
    });
  }

  public render(): React.ReactNode {
    const { onEmailLogin, isFromLogin, isSocialAllowed = true } = this.props;

    return (
      <CommonAuthenticationHoc isFromLogin={isFromLogin} onEmailLogin={onEmailLogin}>
        {!PlatformUtils.isIOS() && isSocialAllowed && this.renderSocialAuths()}
      </CommonAuthenticationHoc>
    );
  }

  private renderSocialAuths = (): React.ReactNode => {
    const { authProviders } = this.state;

    return authProviders.map((auth: SocialAuthProvider) => {
      const initiateSocialLogin = async (): Promise<void> => {
        await this.initiateSocialAuthentication(auth.provider);
      };

      return (
        <>
          {auth.visible && (
            <TouchableOpacity style={styles.alignToCenter} onPress={initiateSocialLogin} key={auth.provider}>
              {this.getImageUrlForLib(auth.provider)}
              <Label type="regular" textType="regular" style={styles.iconTextStyle}>
                {auth.description}
              </Label>
            </TouchableOpacity>
          )}
        </>
      );
    });
  };

  private onSocialAuthSuccess = async (userData: ISocialUserData): Promise<void> => {
    const { loginSuccess, navigation, isFromLogin, onSuccessCallback } = this.props;

    if (navigation) {
      await SocialAuthUtils.onSocialAuthSuccess(
        userData,
        () => {
          navigation.navigate(ScreensKeys.MobileVerification, {
            isFromLogin,
            userData,
          });
        },
        (response) => {
          loginSuccess(response);
          if (onSuccessCallback) {
            onSuccessCallback();
          }
        }
      );
    }
  };

  private initiateSocialAuthentication = async (key: string): Promise<void> => {
    const { isFromLogin, navigation } = this.props;
    this.setLoading(true);
    if (key === SocialAuthKeys.Google) {
      await AuthService.signInWithGoogle(this.onSocialAuthSuccess);
      this.setLoading(false);
      return;
    }

    if (key === SocialAuthKeys.Apple) {
      await AuthService.signInWithApple(this.onSocialAuthSuccess);
      this.setLoading(false);
      return;
    }

    if (key === SocialAuthKeys.Facebook) {
      await AuthService.signInWithFacebook(this.onSocialAuthSuccess, isFromLogin && navigation);
      this.setLoading(false);
    }
  };

  private setLoading = (isLoading: boolean): void => {
    const { toggleLoading } = this.props;
    if (toggleLoading) {
      toggleLoading(isLoading);
    }
  };

  private getImageUrlForLib = (key: string): React.ReactNode => {
    switch (key) {
      case SocialAuthKeys.Facebook:
        return <Facebook />;
      case SocialAuthKeys.Google:
        return <Google />;
      case SocialAuthKeys.LinkedIn:
        return <LinkedIn />;
      case SocialAuthKeys.Apple:
        return <Apple />;
      default:
        return null;
    }
  };
}

export const mapDispatchToProps = (dispatch: Dispatch): IDispatchProps => {
  const { loginSuccess } = UserActions;
  return bindActionCreators(
    {
      loginSuccess,
    },
    dispatch
  );
};

const HOC = connect(
  null,
  mapDispatchToProps
)(withTranslation(LocaleConstants.namespacesKey.auth)(AuthenticationGateways));
export { HOC as AuthenticationGateways };

const styles = StyleSheet.create({
  iconTextStyle: {
    marginTop: 4,
  },
  alignToCenter: {
    alignItems: 'center',
  },
});
