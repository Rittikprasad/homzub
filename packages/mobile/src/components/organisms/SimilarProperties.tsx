import React from 'react';
import { View, StyleSheet, FlatList } from 'react-native';
import { WithTranslation, withTranslation } from 'react-i18next';
import { AssetRepository } from '@homzhub/common/src/domain/repositories/AssetRepository';
import { LocaleConstants } from '@homzhub/common/src/services/Localization/constants';
import { theme } from '@homzhub/common/src/styles/theme';
import { Text } from '@homzhub/common/src/components/atoms/Text';
import PropertyListCard from '@homzhub/mobile/src/components/organisms/PropertyListCard';
import { Asset } from '@homzhub/common/src/domain/models/Asset';

interface IProps {
  propertyTermId: number;
  onSelectedProperty: (propertyTermId: number, propertyId: number) => void;
  transaction_type: number;
}

type Props = WithTranslation & IProps;

interface ISimilarPropertiesState {
  similarProperties: Asset[];
}

export class SimilarProperties extends React.PureComponent<Props, ISimilarPropertiesState> {
  public state = {
    similarProperties: [],
  };

  public componentDidMount = async (): Promise<void> => {
    const { propertyTermId } = this.props;
    await this.getSimilarProperties(propertyTermId);
  };

  public render(): React.ReactNode {
    const { t } = this.props;
    const { similarProperties } = this.state;
    if (similarProperties.length === 0) {
      return null;
    }
    return (
      <View style={styles.container}>
        <Text type="small" textType="semiBold" style={styles.similarProperties}>
          {t('similarProperties')}
        </Text>
        <FlatList
          data={similarProperties}
          horizontal
          showsHorizontalScrollIndicator={false}
          renderItem={this.renderCarouselItem}
          removeClippedSubviews
          keyExtractor={this.renderKeyExtractor}
          testID="similarPropertiesList"
        />
      </View>
    );
  }

  public renderCarouselItem = ({ item }: { item: Asset }): React.ReactElement => {
    const { transaction_type, onSelectedProperty } = this.props;
    const { leaseTerm, saleTerm, id } = item;
    const navigateToSelectedProperty = (): void => {
      if (leaseTerm) {
        onSelectedProperty(leaseTerm.id, id);
      }
      if (saleTerm) {
        onSelectedProperty(saleTerm.id, id);
      }
    };
    return (
      <PropertyListCard
        property={item}
        key={item.id}
        transaction_type={transaction_type}
        isCarousel={false}
        containerStyle={styles.propertyCard}
        textSizeType="small"
        onSelectedProperty={navigateToSelectedProperty}
        testID="listCard"
      />
    );
  };

  private renderKeyExtractor = (item: any, index: number): string => {
    return `${item.id}-${index}`;
  };

  public getSimilarProperties = async (propertyTermId: number): Promise<void> => {
    const { transaction_type } = this.props;
    const response = await AssetRepository.getSimilarProperties(propertyTermId, transaction_type);
    this.setState({ similarProperties: response });
  };
}

export default withTranslation(LocaleConstants.namespacesKey.assetDescription)(SimilarProperties);

const styles = StyleSheet.create({
  container: {
    flex: 1,
    marginVertical: 24,
  },
  similarProperties: {
    color: theme.colors.darkTint4,
  },
  propertyCard: {
    width: theme.viewport.width - 80,
    marginHorizontal: 10,
    shadowColor: theme.colors.shadow,
    shadowOffset: {
      width: 0,
      height: 3,
    },
    shadowOpacity: 0.3,
    shadowRadius: 4.65,
    elevation: 3,
  },
});
