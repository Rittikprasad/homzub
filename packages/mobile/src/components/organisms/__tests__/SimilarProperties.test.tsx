import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import toJson from 'enzyme-to-json';
import { SimilarProperties } from '@homzhub/mobile/src/components/organisms/SimilarProperties';
import { PropertySearchData } from '@homzhub/common/src/mocks/PropertySearchData';

let props: any;
let wrapper: ShallowWrapper;

describe.skip('SimilarProperties', () => {
  const createTestProps = (testProps: any): object => ({
    propertyTermId: 1,
    onFavorite: jest.fn(),
    onSelectedProperty: jest.fn(),
    transaction_type: 1,
    ...testProps,
  });

  props = createTestProps({});
  wrapper = shallow(<SimilarProperties {...props} t={(key: string): string => key} />);

  it('should match snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match snapshot with similar properties', () => {
    wrapper.setState({ similarProperties: PropertySearchData });
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match snapshot for renderItem', () => {
    const RenderItem = wrapper.find('[testID="similarPropertiesList"]').prop('renderItem');
    // @ts-ignore
    const renderItemShallowWrapper = shallow(<RenderItem item={PropertySearchData[0]} />);
    expect(toJson(renderItemShallowWrapper)).toMatchSnapshot();
  });

  it('should call onFavourite', () => {
    const RenderItem = wrapper.find('[testID="similarPropertiesList"]').prop('renderItem');
    // @ts-ignore
    const renderItemShallowWrapper = shallow(<RenderItem item={PropertySearchData[0]} />);
    // @ts-ignore
    renderItemShallowWrapper.find('[testID="listCard"]').prop('onFavorite')();
    expect(props.onFavorite).toBeCalled();
  });

  it('should match snapshot for keyExtractor', () => {
    const KeyExtractor = wrapper.find('[testID="similarPropertiesList"]').prop('keyExtractor');
    // @ts-ignore
    const renderExtractorShallowWrapper = shallow(<KeyExtractor item={PropertySearchData[0]} />);
    expect(toJson(renderExtractorShallowWrapper)).toMatchSnapshot();
  });
});
