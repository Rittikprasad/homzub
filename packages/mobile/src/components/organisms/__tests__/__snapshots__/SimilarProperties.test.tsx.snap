// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimilarProperties should match snapshot 1`] = `""`;

exports[`SimilarProperties should match snapshot for keyExtractor 1`] = `"undefined-[object Object]"`;

exports[`SimilarProperties should match snapshot for renderItem 1`] = `
<withI18nextTranslation(PropertyListCard)
  containerStyle={
    Object {
      "elevation": 3,
      "marginHorizontal": 10,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 3,
        "width": 0,
      },
      "shadowOpacity": 0.3,
      "shadowRadius": 4.65,
      "width": 670,
    }
  }
  isCarousel={false}
  key="1"
  onSelectedProperty={[Function]}
  property={
    Object {
      "asset_group": Object {
        "id": 1,
        "name": "Residential",
      },
      "asset_type": Object {
        "id": 2,
        "name": "Apartment / Condo",
      },
      "block_number": "B",
      "carpet_area": "5000",
      "carpet_area_unit": Object {
        "id": 1,
        "label": "Square feet",
        "name": "SQ_FT",
        "title": "Sq.ft",
      },
      "country": Object {
        "currencies": Array [],
      },
      "floor_number": 1,
      "furnishing": "SEMI",
      "id": 1,
      "images": Array [
        Object {
          "file_name": "House-1a",
          "is_cover_image": true,
          "link": "https://www.investopedia.com/thmb/7GOsX_NmY3KrIYoZPWOu6SldNFI=/735x0/houses_and_land-5bfc3326c9e77c0051812eb3.jpg",
        },
        Object {
          "file_name": "House-1b",
          "is_cover_image": false,
          "link": "https://www.investopedia.com/thmb/7GOsX_NmY3KrIYoZPWOu6SldNFI=/735x0/houses_and_land-5bfc3326c9e77c0051812eb3.jpg",
        },
      ],
      "latitude": "25.61900000",
      "lease_term": Object {
        "currency_code": "INR",
        "expected_price": "20000.0000",
        "id": 1,
      },
      "longitude": "85.12300000",
      "project_name": "Malti Kunj",
      "spaces": Array [
        Object {
          "count": 3,
          "id": 1,
          "name": "Bedroom",
        },
        Object {
          "count": 3,
          "id": 2,
          "name": "Bathroom",
        },
      ],
      "total_floors": 4,
      "unit_number": "41",
    }
  }
  testID="listCard"
  textSizeType="small"
  transaction_type={1}
/>
`;

exports[`SimilarProperties should match snapshot with similar properties 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "marginVertical": 24,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#4C5F72",
      }
    }
    textType="semiBold"
    type="small"
  >
    similarProperties
  </Text>
  <FlatList
    data={
      Array [
        Object {
          "asset_group": Object {
            "id": 1,
            "name": "Residential",
          },
          "asset_type": Object {
            "id": 2,
            "name": "Apartment / Condo",
          },
          "block_number": "B",
          "carpet_area": "5000",
          "carpet_area_unit": Object {
            "id": 1,
            "label": "Square feet",
            "name": "SQ_FT",
            "title": "Sq.ft",
          },
          "country": Object {
            "currencies": Array [],
          },
          "floor_number": 1,
          "furnishing": "SEMI",
          "id": 1,
          "images": Array [
            Object {
              "file_name": "House-1a",
              "is_cover_image": true,
              "link": "https://www.investopedia.com/thmb/7GOsX_NmY3KrIYoZPWOu6SldNFI=/735x0/houses_and_land-5bfc3326c9e77c0051812eb3.jpg",
            },
            Object {
              "file_name": "House-1b",
              "is_cover_image": false,
              "link": "https://www.investopedia.com/thmb/7GOsX_NmY3KrIYoZPWOu6SldNFI=/735x0/houses_and_land-5bfc3326c9e77c0051812eb3.jpg",
            },
          ],
          "latitude": "25.61900000",
          "lease_term": Object {
            "currency_code": "INR",
            "expected_price": "20000.0000",
            "id": 1,
          },
          "longitude": "85.12300000",
          "project_name": "Malti Kunj",
          "spaces": Array [
            Object {
              "count": 3,
              "id": 1,
              "name": "Bedroom",
            },
            Object {
              "count": 3,
              "id": 2,
              "name": "Bathroom",
            },
          ],
          "total_floors": 4,
          "unit_number": "41",
        },
        Object {
          "asset_group": Object {
            "id": 1,
            "name": "Residential",
          },
          "asset_type": Object {
            "id": 2,
            "name": "Apartment / Condo",
          },
          "block_number": "B",
          "carpet_area": "2000",
          "carpet_area_unit": Object {
            "id": 1,
            "label": "Square feet",
            "name": "SQ_FT",
            "title": "Sq.ft",
          },
          "country": Object {
            "currencies": Array [
              Object {
                "currency_code": "",
                "currency_name": "",
                "currency_symbol": "",
              },
            ],
            "id": 1,
            "iso2_code": "",
            "iso3_code": "",
            "name": "",
            "phone_codes": Array [],
          },
          "floor_number": 2,
          "furnishing": "SEMI",
          "id": 2,
          "images": Array [
            Object {
              "file_name": "House-1a",
              "is_cover_image": true,
              "link": "https://www.investopedia.com/thmb/7GOsX_NmY3KrIYoZPWOu6SldNFI=/735x0/houses_and_land-5bfc3326c9e77c0051812eb3.jpg",
            },
            Object {
              "file_name": "House-1b",
              "is_cover_image": true,
              "link": "https://www.investopedia.com/thmb/7GOsX_NmY3KrIYoZPWOu6SldNFI=/735x0/houses_and_land-5bfc3326c9e77c0051812eb3.jpg",
            },
          ],
          "latitude": "25.31400000",
          "lease_term": Object {
            "currency_code": "INR",
            "expected_price": "22000.0000",
            "id": 2,
          },
          "longitude": "85.12350000",
          "project_name": "Mira Sadan",
          "spaces": Array [
            Object {
              "count": 4,
              "id": 1,
              "name": "Bedroom",
            },
            Object {
              "count": 2,
              "id": 2,
              "name": "Bathroom",
            },
          ],
          "total_floors": 3,
          "unit_number": "42",
        },
      ]
    }
    disableVirtualization={false}
    horizontal={true}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onEndReachedThreshold={2}
    removeClippedSubviews={true}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsHorizontalScrollIndicator={false}
    testID="similarPropertiesList"
    updateCellsBatchingPeriod={50}
    windowSize={21}
  />
</View>
`;
